### PR TITLE
[luci] Separate DeadNodeQueryService

### PR DESCRIPTION
**Pull request by seanshpark**
This will separate DeadNodeQueryServiceImpl class to own file

ONE-DCO-1.0-Signed-off-by: SaeHie Park <saehie.park@gmail.com>

### CODE REVIEW

File: compiler/luci/lang/src/DeadNodeQueryService.cpp [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  auto output_nodes = std::set<loco::Node *>(output_nodes_vec.begin(), output_nodes_vec.end());
+  auto active_nodes = loco::active_nodes(output_nodes_vec);
+
+  if (active_nodes.find(node) != active_nodes.end())
+    return false;
+  // input and output nodes are not dead node even if it is not active.
+  if (input_nodes.find(node) != input_nodes.end())
+    return false;
+  if (output_nodes.find(node) != output_nodes.end())
+    return false;
```
Comment (jinevening) [inline compiler/luci/lang/src/DeadNodeQueryService.cpp:56]: It seems that `active_nodes` always contains `output_nodes`. If so, this conditional statement would be redundant.

File: compiler/luci/lang/src/DeadNodeQueryService.cpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    if (circle_node->accept(&d))
+    {
+      assert(node->arity() == 1);
+      loco::Node *real_node = node->arg(0);
+      if (active_nodes.find(real_node) != active_nodes.end())
+        return false;
+      if (input_nodes.find(real_node) != input_nodes.end())
+        return false;
+      if (output_nodes.find(real_node) != output_nodes.end())
+        return false;
```
Comment (jinevening) [inline compiler/luci/lang/src/DeadNodeQueryService.cpp:71]: ditto

File: compiler/luci/lang/src/DeadNodeQueryService.cpp (hunk 1 above)
Comment (seanshpark) [reply compiler/luci/lang/src/DeadNodeQueryService.cpp:56]: This change is to separate to a file and would be better to change in another PR.

Comment (jinevening) [reply compiler/luci/lang/src/DeadNodeQueryService.cpp:56]: Ok. No problem for this PR.

**Pull request merged**